### DISCUSSION TITLE
Macros with flash package

### DISCUSSION THREAD

**Discussion by Geokureli** [HaxeFlixel/flixel#2365] [Q&A]
whenever I try to do macros with my flixel projects I get the error forbidding access to the flash package while in macros. my project isn't targeting flash but openfl classes that recreate flash classes under the guise of `flash.foo`. For instance if I have a macro that uses `ExprOf<FlxPoint>` I get this error because `FlxPoint` imports `flash.geom.Point` instead of the openfl import. 

how do I get around this?

is it worthwhile to do a full project change of flash imports to openfl imports and check for this is codeclimate?

**Comment by RichardBray** [comment]
Do you get this error with any macro or macro's those import Flixel properties? I've used macros in Flixel projects before to manipulate JSON files and that works fine for me.

**Reply by Geokureli** [reply]
Specifically macros involving types that reference the flash package. For instance i had a macro involving a type that imported FlxPoint, which referenced flash.geom.Point (even though it was actually openfl.geom.Point)
